slsDetectorGui/src/qDetectorMain.cpp

```diff
@@ -144,22 +144,22 @@ void qDetectorMain::SetUpWidgetWindow() {
     setFont(QFont("Sans Serif", qDefs::Q_FONT_SIZE, QFont::Normal));
 
     // plot setup
-    myPlot = new qDrawPlot(dockWidgetPlot, myDet);
+    plot = new qDrawPlot(dockWidgetPlot, det);
     FILE_LOG(logDEBUG) << "DockPlot ready";
-    dockWidgetPlot->setWidget(myPlot);
+    dockWidgetPlot->setWidget(plot);
 
     // tabs setup
     tabs = new MyTabWidget(this);
     layoutTabs->addWidget(tabs);
 
     // creating all the other tab widgets
-    tabMeasurement = new qTabMeasurement(this, myDet, myPlot);
-    tabDataOutput = new qTabDataOutput(this, myDet);
-    tabPlot = new qTabPlot(this, myDet, myPlot);
-    tabSettings = new qTabSettings(this, myDet);
-    tabAdvanced = new qTabAdvanced(this, myDet);
-    tabDebugging = new qTabDebugging(this, myDet);
-    tabDeveloper = new qTabDeveloper(this, myDet);
+    tabMeasurement = new qTabMeasurement(this, det, plot);
```

Comment:
Using unique pointers instead of normal pointers then you would not need to delete the object manually in the destructor.

std::uniquie_ptr<qTabMeasurement> = sls::make_unique<qTabMeasurement>(this, det, plot) 
should work.

Reply:
done